providers/implementations/keymgmt/rsa_kmgmt.c

```diff
@@ -454,19 +454,25 @@ static void *gen_init(void *provctx, int selection, int rsa_type,
         gctx->libctx = libctx;
         if ((gctx->pub_exp = BN_new()) == NULL
             || !BN_set_word(gctx->pub_exp, RSA_F4)) {
-            BN_free(gctx->pub_exp);
-            OPENSSL_free(gctx);
-            return NULL;
+            goto err;
         }
         gctx->nbits = 2048;
         gctx->primes = RSA_DEFAULT_PRIME_NUM;
         gctx->rsa_type = rsa_type;
-    }
+    } else
+        goto err;
+
     if (!rsa_gen_set_params(gctx, params)) {
```

Comment:
Style nit: no {} around this one.

The guide is that single statements don't need {} but both branches of an _if_ should be the same.

Reply:
Done. Thanks for your reminder.